modules/programs/powerline-go.nix

```diff
@@ -0,0 +1,120 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.powerline-go;
+
+  # Convert an option value to a string to be passed as argument to
+  # powerline-go:
+  valueToString = value:
+    if builtins.isList value then
+      builtins.concatStringsSep "," (builtins.map valueToString value)
+    else if builtins.isAttrs value then
+      valueToString
+      (mapAttrsToList (key: val: "${valueToString key}=${valueToString val}")
+        value)
+    else
+      builtins.toString value;
+
+  modulesArgument = optionalString (cfg.modules != null)
+    "-modules ${valueToString cfg.modules}";
+
+  newlineArgument = optionalString cfg.newline "-newline";
+
+  pathAliasesArgument = optionalString (cfg.pathAliases != null)
+    "-path-aliases ${valueToString cfg.pathAliases}";
+
+  otherSettingPairArgument = name: value:
+    if value == true then "-${name}" else "-${name} ${valueToString value}";
+
+  otherSettingsArgument = optionalString (cfg.settings != { })
+    (concatStringsSep " "
+      (mapAttrsToList otherSettingPairArgument cfg.settings));
+
+  commandLineArguments = ''
+    ${modulesArgument} ${newlineArgument} ${pathAliasesArgument} ${otherSettingsArgument}
+  '';
+
+in {
+  meta.maintainers = [ maintainers.DamienCassou ];
+
+  options = {
+    programs.powerline-go = {
+      enable = mkEnableOption "Powerline-go";
```

Comment:
Maybe this could be expanded, so one does not have to google what powerlien is?

Reply:
done.